### PR TITLE
Update Python API for Card selection

### PR DESCRIPTION
**Pull request by mmeldo**
Currently, only possible API call is to add a Card to selection. I have added an option to clear the selection before the Card is added. To have the Card selection API complete, I have also added call to remove the card from selection.

### CODE REVIEW

**Comment by iGemini** [comment]
I have to wonder if it makes more sense to have ```select``` itself act as a toggle, rather than have a separate function to deselect (or unselect?) a card.

**Comment by mmeldo** [comment]
I did not want to change the current method to be compatible.
Ideally, it would be best to have one method to get the selected state, one to clear the selection and then either:
* toggle `select` method
* setting `setSelection(boolean)` method 
Toggle would mean finding a card in selection, but would be probably more neat. Honestly, for me either way works. Maybe for 3.1.0.3?

**Comment by mmeldo** [comment]
I implemented changes as suggested by @brine : 

> card.select() to add selection, card.select(false) to remove selection
then just have a global function to clear the selection list

the convention is to use a getter and setter, but the select method does not work directly with the _select property, it works with the Selection instead so I guess that is OK.

**Comment by brine** [comment]
yeah we've been upgrading a lot of these to property setters, I suppose this one missed the sweep in 3.1.0.2.

This looks like the best way to handle it right now while still respecting backwards compatibility.

File: octgnFX/Octgn/Scripting/PythonAPI.py [hunk 1]
```diff
@@ -227,7 +230,7 @@ def setIndex(self, index):
     _api.CardSetIndex(self._id,index, True)
   @property
   def getIndex(self): return _api.CardGetIndex(self._id)
-  def select(self): _api.CardSelect(self._id)
+  def select(self, selection = True): _api.CardSelect(self._id, selection)
   def peek(self): _api.CardPeek(self._id)
   def target(self, active = True): _api.CardTarget(self._id, active)
   def arrow(self, targetCard, active = True): _api.CardTargetArrow(self._id, targetCard._id, active)
```
[the remark this answers is not in the document]
Comment (mmeldo) [reply octgnFX/Octgn/Scripting/PythonAPI.py:236]: I do not think it is referenced. At least when I tested, it was not used and actually is not up to date with the latest version of API. I changed it just to be safe.

**Comment by mmeldo** [comment]
Any idea why clahub check is pending? And sorry, I do not want to be annoying, but I need this update for one of my new game features. Do you have at least some estimate when it can be published? Thanks.

**Comment by brine** [comment]
not sure, the merges may have done something to the license agreement verification.  I believe we were holding on pull requests until the server maintenance was finished, so I'll check with the boss tonight

**Pull request merged**